src/data.ts

```diff
@@ -139,7 +141,41 @@ export function isNamedData(data: Partial<Data> | Partial<VgData>): data is Name
   return !!data['name'] && !isUrlData(data) && !isInlineData(data);
 }
 
+export function isGenerator(data: Partial<Data> | Partial<VgData>): data is Generator {
+  return data && isGraticuleGenerator(data);
+}
+
+export function isGraticuleGenerator(data: Partial<Data> | Partial<VgData>): data is GraticuleGenerator {
+  return !!data['graticule'];
+}
+
 export type DataSourceType = 'raw' | 'main' | 'row' | 'column' | 'lookup';
 
 export const MAIN: 'main' = 'main';
 export const RAW: 'raw' = 'raw';
+
+export type Generator = GraticuleGenerator;
+
+export interface GeneratorBase {
+  /**
+   * Provide a placeholder name and bind data at runtime.
+   */
+  name?: string;
+}
+
+export interface GraticuleGenerator extends GeneratorBase {
+  /**
+   * Generate graticule GeoJSON data for geographic reference lines.
+   */
+  graticule: true | GraticuleParams;
+}
+
+export interface GraticuleParams {
```

Comment:
If we want the TSDocs for each of the property to be exactly the same as ones in `vega-typing`'s `VgGraticuleTransform` (which has no docs right now), we can do. 

```
export type GraticuleParams = Omit<VgGraticuleTransform, 'type'>;
```

(We can also redefine docs for jus some properties too.) 

```
export type GraticuleParams = Omit<VgGraticuleTransform, 'type'> & {
  /** different docs text*/
  extent?: number[][];
};
```

Reply:
I tried this, and while it worked within VL proper, ts-json-schema-generator crashed as a result:

`UnknownNodeError: Unknown node "Exclude<keyof T, K>" (ts.SyntaxKind = 175) at /work/vega/vega-lite/src/util.ts(326,49)`

Reply:
https://github.com/vega/ts-json-schema-generator/issues/93